projects/kit/components/input-phone-international/input-phone-international.component.ts

```diff
@@ -0,0 +1,252 @@
+            align: 'right',
+        }),
+    ],
+})
+export class TuiInputPhoneInternationalComponent
+    extends AbstractTuiControl<string>
+    implements TuiFocusableElementAccessor
+{
+    /* eslint-disable @typescript-eslint/member-ordering */
+    protected readonly options = inject(TUI_INPUT_PHONE_INTERNATIONAL_OPTIONS);
+    protected readonly countryIsoCode$ = new BehaviorSubject(this.options.countryIsoCode);
+    protected readonly mask$ = new BehaviorSubject<MaskitoOptions | null>(null);
+    protected readonly countriesNames$ = inject(TUI_COUNTRIES);
+    protected readonly textfieldSize = inject(TUI_TEXTFIELD_SIZE);
+
+    protected open = false;
+    protected textfieldValue = '';
+
+    @ViewChild(TuiPrimitiveTextfieldComponent)
+    private readonly inputPhone?: TuiPrimitiveTextfieldComponent;
+
+    @ViewChild(TuiTextfieldComponent)
+    private readonly countrySelect?: TuiTextfieldComponent<unknown>;
+
+    @Input()
+    public countries = this.options.countries;
+
+    @Output()
+    public readonly countryIsoCodeChange = this.countryIsoCode$.pipe(
+        distinctUntilChanged(),
+        skip(1),
+    );
+
+    @Input('countryIsoCode')
+    public set isoCode(code: TuiCountryIsoCode) {
+        this.countryIsoCode$.next(code);
+    }
+
+    public get nativeFocusableElement(): HTMLElement | null {
+        return this.inputPhone && !this.computedDisabled
+            ? this.inputPhone.nativeFocusableElement
+            : null;
+    }
+
+    public get focused(): boolean {
+        return Boolean(this.countrySelect?.focused || this.inputPhone?.focused);
+    }
+
+    public onPaste(event: Event, phonesMetadata: MetadataJson | null): void {
+        if (
+            !tuiIsInputEvent(event) ||
+            !phonesMetadata ||
+            (!event.inputType.includes('Drop') && !event.inputType.includes('Paste'))
+        ) {
+            return;
+        }
+
+        const newValue = event.data || '';
+        const prefixedValue = newValue.startsWith(CHAR_PLUS)
+            ? newValue
+            : CHAR_PLUS + newValue;
+
+        if (validatePhoneNumberLength(prefixedValue) === 'TOO_SHORT') {
+            return;
+        }
+
+        const countryIsoCode = maskitoGetCountryFromNumber(prefixedValue, phonesMetadata);
+
+        if (countryIsoCode) {
+            this.countryIsoCode$.next(countryIsoCode);
+        }
+    }
+
+    public onItemClick(isoCode: TuiCountryIsoCode): void {
+        this.open = false;
+        this.countryIsoCode$.next(isoCode);
+        this.nativeFocusableElement?.focus();
+    }
+
+    constructor() {
+        super();
+        combineLatest([this.countryIsoCode$, this.options.metadata])
+            .pipe(
+                map(args => this.computeMask(...args)),
+                takeUntilDestroyed(),
+            )
+            .subscribe(this.mask$);
+    }
+
+    public override setDisabledState(): void {
+        super.setDisabledState();
+        this.close();
```

Comment:
I think we can get rid of this by adding `!computedDisabled` to the [tuiDropdownOpen], or maybe come up with some universal for this later, as we will need it in every control with a dropdown.

Reply:
Fixed in https://github.com/taiga-family/taiga-ui/pull/7580/commits/006c66da85e757dc13595046e600d6255e5873c1